### ISSUE TITLE
Always "No file changes detected" on v2.4.2

### ISSUE THREAD

**Issue by at-wat** [repo-sync/pull-request#45]
The action doesn't create PR even if there are commits between destination and source branch.
```
No file changes detected between source and destination branches.
```

It looks be caused by #43 which changed diff command to
```
git diff --name-only -- "$DESTINATION_BRANCH" "$SOURCE_BRANCH"
```

Below shows the change of the behavior by #43:
```shell
$ git init
Initialized empty Git repository in /tmp/pull-request/.git/

# Add a.txt to master branch
$ touch a.txt; git add a.txt; git commit -m "Add a.txt"
[master (root-commit) 5bd1d04] Add a.txt
 1 file changed, 0 insertions(+), 0 deletions(-)
 create mode 100644 a.txt

# Create the-branch and add b.txt, then master and the-branch should have diff
$ git checkout -b the-branch
Switched to a new branch 'the-branch'
$ touch b.txt; git add b.txt; git commit -m "Add b.txt"
[the-branch 27ce7e0] Add b.txt
 1 file changed, 0 insertions(+), 0 deletions(-)
 create mode 100644 b.txt
$ DESTINATION_BRANCH=master
$ SOURCE_BRANCH=the-branch

# This is the script after #43 and LINES_CHANGED is 0 (**incorrect**)
$ LINES_CHANGED=$(git diff --name-only -- "$DESTINATION_BRANCH" "$SOURCE_BRANCH" | wc -l | awk '{print $1}')
$ echo $LINES_CHANGED 
0

# This is the script before #43 and LINES_CHANGED is 1 (**correct**)
$ LINES_CHANGED=$(git diff --name-only "$DESTINATION_BRANCH" "$SOURCE_BRANCH" | wc -l | awk '{print $1}')
$ echo $LINES_CHANGED 
1
```

**Comment by at-wat** [comment]
I think the current script takes diff between the index and the worktree. 
> ```
>        git diff [<options>] [--] [<path>...]
>            This form is to view the changes you made relative to the index (staging area
>            for the next commit). In other words, the differences are what you could tell
>            Git to further add to the index but you still haven’t. You can stage these
>            changes by using git-add(1).
> ```

It should be fixed to the following form.

> ```
>        git diff [<options>] <commit> <commit> [--] [<path>...]
>            This is to view the changes between two arbitrary <commit>.
> ```

**Comment by at-wat** [comment]
```
# The script fixed by #46
$ LINES_CHANGED=$(git diff --name-only "$DESTINATION_BRANCH" "$SOURCE_BRANCH" -- | wc -l | awk '{print $1}')
$ echo $LINES_CHANGED 
1
```
This form of git diff command should be correct for this case.
